### PR TITLE
EVG-2613 Add individual task execution data support in the REST v2 API

### PR DESCRIPTION
**Pull request by richardsamuels**

### CODE REVIEW

File: model/task/task_test.go [hunk 1]
```diff
@@ -964,3 +964,26 @@ func TestMergeTestResultsBulk(t *testing.T) {
 	}
 	assert.Equal(3, count)
 }
+
+func TestFindOldTasksByID(t *testing.T) {
+	assert := assert.New(t) //nolint
+
```
Comment (john-m-liu) [inline model/task/task_test.go]: clear the collection beforehand

File: rest/data/task.go [hunk 2]
```diff
@@ -90,6 +90,15 @@ func (tc *DBTaskConnector) FindTasksByIds(ids []string) ([]task.Task, error) {
 	return ts, nil
 }
 
+func (tc *DBTaskConnector) FindOldTasksByID(id string) ([]task.Task, error) {
+	ts, err := task.FindOld(task.ByIDPrefix(id))
```
Comment (john-m-liu) [inline rest/data/task.go]: Why does this just find old executions? Why not search both collections?

Comment (richardsamuels) [reply rest/data/task.go]: I was under that impression that only the most recent execution is stored in tasks, and all the older ones are stored in old_tasks. I only want the old executions. Was this assumption wrong?

Comment (john-m-liu) [reply rest/data/task.go]: That's correct, I didn't realize that the caller of this retrieved the new and old versions separately

**Pull request merged**